### DISCUSSION TITLE
How to model a dynamic heterogeneous attributed graph dataset in PyG

### DISCUSSION THREAD

**Discussion by kamizhou** [pyg-team/pytorch_geometric#7964] [Q&A]
I‘ve noticed there are `HeteroData()` and `TemporalData()` in `torch_geometric.data` , but how could I combine them to handle or create a dynamic heterogeneous directed attributed graph in PyG ? 
For example, TypeA node `NA1` with attributes `VA1` has a TypeM connection `EM1` with attributes `VM1` with TypeB node `NB1` with attributes `VB1`  in timestamp `T1`, while TypeB node `NB2` with attributes `VB2` has a TypeN connection `EN2` with attributes `VN2` with TypeC node `NC3`  with attributes `VC3` regardless of timestamp.

**Comment by rusty1s** [comment]
You can add your timestamp to the node types of a `HeteroData` object and then use, e.g., `NeighborLoader` to process that information. `NeighborLoader` supports temporal sampling via `input_time` and `time_attr` such that you only sample nodes that exist in time.
